codexctl/device.py

```diff
@@ -274,11 +286,140 @@ def connect_to_device(
 
         return client
 
-    def get_device_status(self) -> tuple[str | None, str, str]:
+    def _read_version_from_path(self, ftp, base_path: str = "") -> tuple[str, bool]:
+        """Reads version from a given path (current partition or mounted backup)
+
+        Args:
+            ftp: SFTP client connection
+            base_path: Base path prefix (empty for current partition, /tmp/mount_pX for backup)
+
+        Returns:
+            tuple: (version_string, old_update_engine_boolean)
+        """
+        old_update_engine = True
+
+        try:
+            update_conf_path = f"{base_path}/usr/share/remarkable/update.conf" if base_path else "/usr/share/remarkable/update.conf"
+            with ftp.file(update_conf_path) as file:
+                version = re.search(
+                    "(?<=REMARKABLE_RELEASE_VERSION=).*",
+                    file.read().decode("utf-8").strip("\n"),
+                ).group()
+        except Exception:
+            os_release_path = f"{base_path}/etc/os-release" if base_path else "/etc/os-release"
+            with ftp.file(os_release_path) as file:
+                version = (
+                    re.search("(?<=IMG_VERSION=).*", file.read().decode("utf-8"))
+                    .group()
+                    .strip('"')
+                )
+                old_update_engine = False

```

Comment:
This should really be updated to not blindly try the next mechanism on any exception, and instead check for the files, and that they have the expected content.